### PR TITLE
docs: Cluster-Wide Permissions and Operator Deployment

### LINKED ISSUE #4833: [Docs]: Cluster-Wide Permissions and Operator Deployment
### Is there an existing issue already for your request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new documentation enhancement to be evaluated.

### What problem in the existing documentation this issue aims to solve?

This ticket includes the proposed new section on "Cluster-Wide Permissions and Operator Deployment" from PR #4638. I believe this section is out of place (located at the bottom of the page) and seems to have been written without fully understanding the content on the page.

I think this section should be aligned with the existing RBAC section, integrated with it, and most importantly, should not contradict it.

### Describe what additions need to be done to the documentation

Describe cluster-wide permissions and operator deployments

### Describe what pages need to change in the documentation, if any

security.md

### Describe what pages need to be removed from the documentation, if any

none

### Additional context

see #4638 

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gbartolini**
Closes #4833

### CODE REVIEW

**Review by leonardoce — APPROVED** [review]

File: docs/src/security.md [hunk 1]
```diff
@@ -169,11 +169,59 @@ namespaced resources.
   decide in which nodes a pod can be scheduled preventing the replicas to be
   in the same node, specially if nodes are in different availability zones. This
   permission is also used to determine if a node is schedule or not, avoiding
-  the creation of pods that cannot be created at all.
+  the creation of pods that cannot be created at all and/or trigger a switch over if
```
Comment (jsilvela) [inline docs/src/security.md:172]: hmm, this is not very good. Going to try some changes

```suggestion
  the creation of pods that cannot be created at all and/or trigger a switch over if
```

File: docs/src/security.md [hunk 2]
```diff
@@ -165,15 +165,66 @@ namespaced resources.
...
+
+Even with these permissions, if someone gains access to the `ServiceAccount`,
+they will only have `get`, `list`, and `watch` permissions, which are limited
+to viewing resources. If an unauthorized user already has access to the
+`ServiceAccount`, it indicates a more significant security issue.
+
+It's crucial to prevent users from accessing the operator's `ServiceAccount`
+and any other `ServiceAccount` with elevated permissions.
+
+#### Can This Issue Be Mitigated?
```
Comment (jsilvela) [inline docs/src/security.md:216]: Don't like this section. Firstly, what "Issue" is it making reference to?
Also, it looks way too apologetic.

File: docs/src/security.md [hunk 3]
```diff
@@ -169,14 +169,13 @@ namespaced resources.
...
   in the same node, specially if nodes are in different availability zones. This
   permission is also used to determine if a node is schedule or not, avoiding
-  the creation of pods that cannot be created at all.
+  the creation of pods that cannot be created at all and/or trigger a switch over if
+  the primary lives in a unscheduled node.
 
 To see all the permissions required by the operator, you can run `kubectl
 describe clusterrole cnpg-manager`.
 
-<!-- TODO: please integrate this or align it with the RBAC section -->
```
Comment (gbartolini) [inline docs/src/security.md]: Why was this not considered? @sxd

Comment (sxd) [reply docs/src/security.md]: It was

**Comment by gbartolini** [comment]
@sxd @jsilvela can you please verify this is fine?

File: docs/src/security.md [hunk 4]
```diff
@@ -165,14 +171,63 @@ namespaced resources.
...
+```sh
+kubectl describe clusterrole cnpg-manager
+```
+
+!!! Info
+    You can customize the YAML file to align with your security requirements by
+    using a `RoleBinding` instead of a `ClusterRoleBinding`. This approach enables
+    deployment of the operator in a dedicated namespace with restricted access from
+    administrators. This ensures that standard users cannot access the operator's
+    `ServiceAccount`.
```
Comment (jsilvela) [inline docs/src/security.md:204]: I don't think we explain anywhere how to do this, and it's not obvious.
Should we have this info box?
Or maybe add a section on how to customize the YAML?

**Review by jsilvela — APPROVED** [review]

**Comment by jsilvela** [comment]
Aside from my comment on the info box, looks good.

**Review by sxd — APPROVED** [review]

File: docs/src/security.md (hunk 4 above)
Comment (sxd) [reply docs/src/security.md:204]: Probably we can do that on the plugin side, but that will be out of the scope of this documentation for sure

Comment (gbartolini) [reply docs/src/security.md:204]: I found this awkward too, but I understand the need to outline this possibility. I have tried several times to write a sentence that explains that CNPG doesn't support this yet, but could not come with an easy and clear way to do it. Hence, less is more and decided to omit it.

Comment (gbartolini) [reply docs/src/security.md:204]: Honestly, I was expecting the whole `cnpg generate` command to already take into account all of this, and was puzzled when I discovered it wasn't actually working. So I'd rather work on a solution than documenting the current state (which is not a feature we provide).

**Pull request merged**